src/odemis/acq/fastem.py

```diff
@@ -80,18 +86,136 @@ def estimateTime(roas):
+    # FIXME: if the stream has .horizontalFoV, it's automatically set to the biggest
+    # value, which do not work.
+
+    # To avoid the focuser, and horizontalFoV, and the already present data, we
+    # just create our own SEMStream
+    sem_stream = SEMStream(stream.name.value + " copy", stream.detector, stream.detector.data, stream.emitter)
+
+    est_dur = estimateTiledAcquisitionTime(sem_stream, stage, area)
+    f = model.ProgressiveFuture(start=time.time(), end=time.time() + est_dur)
+    _executor.submitf(f, _run_overview_acquisition, f, sem_stream, stage, area, live_stream)
+
     return f
 
 
-def _run_fake_overview_acquisition(stream, stage, area, live_stream):
+def estimateTiledAcquisitionTime(stream, stage, area):
+    # TODO: fix function to limit the acquisition area so that the FoV is taken into account.
+    # t_estim = estimateTiledAcquisitionTime(stream, stage, area, overlap=0)
+
+    # For now, it's just simpler to hard-code the time spent per tile, and derive the total time based on it.
+    fov = (TILE_FOV_X, TILE_FOV_X * TILE_RES[1] / TILE_RES[0])
+    normalized_area = util.normalize_rect(area)
+    area_size = (normalized_area[2] - normalized_area[0],
+                 normalized_area[3] - normalized_area[1])
+    nx = math.ceil(abs(area_size[0] / fov[0]))
+    ny = math.ceil(abs(area_size[1] / fov[1]))
+
+    # TODO: compensate for longer dwell times => should be a A+Bx formula?
+    return nx * ny * TIME_PER_TILE_1US  # s
+
+
+def _configure_overview_hw(scanner):
+    """
+    Set good parameters an overview acquisition
+    """
+    # Typically, all these settings should have already been set, but better be safe.
+    scanner.multiBeamMode.value = False
+    scanner.external.value = False
+    scanner.blanker.value = None  # Automatic
+    # Disable immersion, to get a larger field of view.
+    # It needs to be done before changing the horizontalFoV, as the range is updated
+    scanner.immersion.value = False
+
+    scanner.horizontalFoV.value = TILE_FOV_X  # m
+
+    # => compute the scale needed in X, use the same one in Y, and then compute the Y resolution.
+    # => set resolution to full FoV, and then adjust
+    scale = scanner.shape[0] / TILE_RES[0]
+    scanner.scale.value = (scale, scale)
+    if scanner.resolution.value != TILE_RES:
+        logging.warning("Unexpected resolution %s on e-beam scanner, expected %s",
+                        scanner.resolution.value, TILE_RES)
+
+
+def _run_overview_acquisition(f, stream, stage, area, live_stream):
     """
-    :param coords: (float, float, float, float) minx, miny, maxx, maxy coordinates of overview region
-    :param sem_stream (SEMStream):
     :returns: (DataArray)
     """
-    time.sleep(2)
-    d = model.DataArray(numpy.random.random((500, 500)))
-    return d
+    _configure_overview_hw(stream.emitter)
+
+    overlap = STAGE_PRECISION / stream.emitter.horizontalFoV.value
+    logging.debug("Overlap is %s%%", overlap * 100)  # normally < 1%
```

Comment:
Is this overlap enough for stitching? For the identity registrar it shouldn't be a problem. Are we planning on using real registration in the future? Or does that not make sense?

Reply:
Yes! It's sufficient for stitching... assuming that you don't need to do registration, because we can read the stage position very precisely. That's exactly our case. The pixel size is so big compared to the precision of the position reading that the tiles are always at the right place (within a pixel). 
(It's not yet exactly correct now, as the SEM image size needs to be calibrated)

Reply:
I've add some comment to explain that.